shared/src/main/scala/StringHelpers.scala

```diff
@@ -15,6 +16,75 @@ object StringHelpers:
       case a: VNum  => a.toInt.toChar.toString
       case a: VList => VList(a.map(chrord)*)
 
+  // https://codegolf.stackexchange.com/a/151721/78850
+  def compressDictionary(s: String)(using ctx: Context): String =
+    val endLength = 2 + ctx.globals.longDictionary.map(_.length).max
+
+    def character(z: Int, c: Char) =
+      val o =
+        if c.toInt == 10 then 95
+        else if ' ' <= c && c <= '~' then c.toInt - 32
+        else throw new Exception(s"Invalid character $c")
+
+      3 * 96 * z + o
+
+    def dictionary(z: Int, w: String, nonempty: Boolean) =
+      var ts = nonempty
+      var subW = w
+      if w.headOption.exists(_ == ' ') then
+        subW = w.substring(1)
+        ts = !ts
+      val dictionary =
+        if w.size < 6 then ctx.globals.shortDictionary
+        else ctx.globals.longDictionary
+      val swapcase = !dictionary.exists(_ == w)
+      val word =
+        if !swapcase then w
+        else
+          val first = w.charAt(0)
+          val rest = w.substring(1)
+          val firstToggled =
+            if first.isUpper then first.toLower else first.toUpper
+          firstToggled.toString + rest
+
+      if !dictionary.contains(word) then
+        throw new Exception(s"Invalid word $word")
+      val f = ts || swapcase
+      val j = if swapcase then 2 else 1
+      val i = dictionary.indexOf(word)
+
+      var res = dictionary.size * z + i
+      if f then
+        res = 3 * res + j
+        res = 3 * res + 2
+      else res = 3 * res + 1
+      res
+    end dictionary
+
+    def go(z: Int) =
+      val compressed = StringBuilder()
+      var z1 = z
```

Comment:
Can't you make the var/method names a bit more... descriptive?

Reply:
This was ported directly from what Dennis wrote. Someday, we mere mortals will be able to decipher what he wrote. Until then, we'll have to stick with cryptic single-letter names :|

Reply:
`z1` is the number that represents the string, to be converted into our compressed string (you can see the divmod by 250, which is the base 250 compression).

Reply:
This whole `go` function is just number compression. Could be outsorced to a helper. Also idk our literal syntax but we can probably use a higher base than 250.

Reply:
I tried changing the base to 252, but it seemed to break compression. Maybe I did something wrong, or maybe the modulos need to change as well

Reply:
You have to change both 250s, and you might also have to mess with the codepage it's using for indexing, idk